k4MarlinWrapper/src/components/Lcio2EDM4hep.cpp

```diff
@@ -208,9 +219,19 @@ StatusCode Lcio2EDM4hepTool::convertCollections(lcio::LCEventImpl* the_event) {
   }
 
   // Set the ParticleID meta information
-  auto& metadataFrame = m_podioDataSvc->getMetaDataFrame();
-  for (const auto& [collName, pidInfo] : pidInfos) {
-    edm4hep::utils::PIDHandler::setAlgoInfo(metadataFrame, collName, pidInfo);
+  // TODO: Clean up
+  if (m_podioDataSvc) {
+    auto& metadataFrame = m_podioDataSvc->getMetaDataFrame();
+    for (const auto& [collName, pidInfo] : pidInfos) {
+      edm4hep::utils::PIDHandler::setAlgoInfo(metadataFrame, collName, pidInfo);
+    }
+  } else {
+    for (const auto& [collName, pidInfo] : pidInfos) {
+      k4FWCore::putParameter(podio::collMetadataParamName(collName, edm4hep::labels::PIDAlgoName), pidInfo.algoName);
+      k4FWCore::putParameter(podio::collMetadataParamName(collName, edm4hep::labels::PIDAlgoType), pidInfo.algoType());
+      k4FWCore::putParameter(podio::collMetadataParamName(collName, edm4hep::labels::PIDParameterNames),
+                             pidInfo.paramNames);
```

Comment:
I would have a preference of doing this via https://github.com/key4hep/k4FWCore/pull/273 as we would not leak all of these details everywhere, which would also minimize the risk of breaking things accidentally because no one remembers where all the usages are.

Reply:
Done